### ISSUE TITLE
Fortran runtime error

### ISSUE THREAD

**Issue by florianhartig** [trotsiuk/r3PG#1]
I can compile the package, but when I try running the example I get 

`Fortran runtime error: Funny sized logical array`

I assume it's a problem with my fortran version or something similar, but not sure. Any idea?

**Comment by trotsiuk** [comment]
The only one time I'm using type `logical` is [here](https://github.com/trotsiuk/r3PGmix/blob/9a87b38092579ad858e2de32765580083ceed24c/pkg/src/md_3PG.f95#L707-L725). 
I would look and read more about different compilers. Do you know which fortran compiler you have?

**Comment by florianhartig** [comment]
I think gfortran 8.2 ... it doesn't work with the official 4.x version from R.

**Comment by trotsiuk** [comment]
But can you compile, or even buil/compiling doesn't work?

**Comment by trotsiuk** [comment]
Can you test if below mentioned example from [here](https://stackoverflow.com/questions/58354499/calling-fortran-from-r-funny-sized-logical-array) will give some errors

save as `test_logical.f95`

```
subroutine fun(n, x)
  implicit none
  integer, intent(in) :: n
  logical, intent(in) :: x(n)
  integer :: i

  write(*,*) pack([(i, i=1, n)], x)

end subroutine fun
```

And run this from R

```
system("gfortran -shared  -fPIC -o test_logical test_logical.f95")

dyn.load("test_logical")
x <- c(TRUE ,FALSE)
f <- .Fortran("fun", 2L, as.logical(x))


# and this

system("R CMD SHLIB test_logical.f95")
dyn.load("test_logical.so")
x <- c(TRUE ,FALSE)
f <- .Fortran("fun", 2L, as.logical(x))
```

**Comment by trotsiuk** [comment]
Hi @florianhartig 
I wonder if you have some update on your issue with `fortran`. Or no success so far?

**Comment by florianhartig** [comment]
OK, I am still having problems, I don't understand it ... will keep on trying, maybe a problem with my Xcode. Which MacOS are you running, I am still on Mojave.

**Comment by florianhartig** [comment]
On the plus side, I have finally managed to get this compiled on my computer. I updated to Catalina, updated XCode, Clang, which still didn't help, then erased all my old gFortran compilers and tried 4, 6 (from R), didn't work, and finally 8 from GNU worked. Maybe the 8 thing was the problem all the time - I had thought I'm running gfortran 8, but in the end saw that I still had a few older versions on my mac.
